File: Lab5C++/TVector2.cpp
#include "TVector2.h"
#include<math.h>
TVector2::TVector2(float x, float y)
{
	this->x = x;
	this->y = y;
}
float TVector2::get_coord(char coord)
{
	switch (coord)
	{
	case 'x':
		return x;
		break;
	case 'y':
		return y;
		break;
	}
}
float TVector2::get_length()
{
	return sqrt(x * x + y * y);
}
float TVector2::scalar(TVector& other)
{
	return x*other.get_coord('x')+y*other.get_coord('y');
}

File: Lab5Py/TVector2.py
from TVector import TVector
from math import sqrt

class TVector2(TVector):
    def __init__(self, x=0, y=0):
        self.__x=x 
        self.__y=y
    def get_length(self):
        return sqrt(self.__x*self.__x+self.__y*self.__y)
    def get_coord(self, coord):
            return self.__x if coord=='x' else self.__y
    def scalar(self, other):
        return self.__x*other.get_coord('x')+self.y*other.get_coord('y')

    
   




